src/Transactions.sol

```diff
@@ -38,7 +53,26 @@ library Transactions {
         return RLPWriter.writeList(items);
     }
 
-    function decodeRLP(bytes memory rlp) internal pure returns (Legacy memory) {
+    function encodeRLP(EIP1559 memory txStruct) internal pure returns (bytes memory) {
+        bytes[] memory items = new bytes[](12);
```

Comment:
Is this encoding correct? `EIP-1559` transactions are encoded with [`EIP-2718`](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2718.md) format, so it needs to have the transaction type prefixed. You can find a code reference [here](https://github.com/Lilyjjo/tldr-research/blob/main/poc_only_suave/suave-contracts/src/SuaveSigner.sol#L128).

Reply:
I just fixed this, lmk if it is still not correct.